source/slang/slang-mangle.cpp

```diff
@@ -547,6 +547,28 @@ namespace Slang
 
             for(auto paramDeclRef : parameters)
             {
+                // parameter modifier makes big difference in the spirv code generation, for example
+                // "out"/"inout" parameter will be passed by pointer. Therefore, we need to
+                // distinguish them in the mangled name to avoid name collision.
+                ParameterDirection paramDirection = getParameterDirection(paramDeclRef.getDecl());
+                switch (paramDirection)
+                {
+                case kParameterDirection_Ref:
+                    emitRaw(context, "ref");
```

Comment:
Let's try to make mangled names shorter by shortening those prefixes. in can be i_, inout can be io_, out can be o_, ref can be r_, const ref can be cr_.

Reply:
The purpose of mangling is to encode info in a compact identifier string. Readability is a secondary goal. We should also try to make sure the encoding scheme has a viable decoding scheme, so if needed we can parse the mangled name back to the structural representation about a function.

Reply:
done